plugins/snowflake/dbt/adapters/snowflake/connections.py

```diff
@@ -110,27 +108,22 @@ def open(cls, connection):
             return connection
 
         try:
-            credentials = connection.credentials
-            # Pull all of the optional authentication args for the connector,
-            # let connector handle the actual arg validation
-            auth_args = {auth_key: credentials[auth_key]
-                         for auth_key in ['user', 'password', 'authenticator']
-                         if auth_key in credentials}
-
-            auth_args['private_key'] = cls._get_private_key(
-                credentials.get('private_key_path'),
-                credentials.get('private_key_passphrase'))
+            creds = connection.credentials
 
             handle = snowflake.connector.connect(
-                account=credentials.account,
-                database=credentials.database,
-                schema=credentials.schema,
-                warehouse=credentials.warehouse,
-                role=credentials.get('role', None),
+                account=creds.account,
+                # TODO: the previous code tried to make it optional as of the
+                # addition of  Snowflake, but 'user' is required by the schema,
+                # so it was always present. Should it be optional and be part
+                # of auth_args?
```

Comment:
@drewbanin please check my reading of the old logic here, but I think `user` was effectively always required because it's in the "required" fields of the json schema. Which seems reasonable enough to me, but maybe we shouldn't be requiring `user` at all?